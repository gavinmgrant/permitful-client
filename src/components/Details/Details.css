.details {
    position: fixed;
    bottom: 0;
    width: 100%;
    background-color: #18181E;
    color: #ffffff;
    padding: 10px;
}

.details-block {
    margin: 10px;
    padding: 10px;
    color: #18181E;
    background-color: #DEC79B;
    border-radius: 5px;
}

.details-address {
    margin-top: 0;
}

@media screen and (min-width: 600px) {
    .details {
        padding: 15px;
    }
}

@media screen and (min-width: 1024px) {
    .details {
        width: 1024px;
    }

    .details-block {
        margin: 20px;
        padding: 20px;
    }
}