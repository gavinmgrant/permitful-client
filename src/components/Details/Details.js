import React from 'react';
import './Details.css';

export default function Details(props) {
    const date = props.statusDate
    const formattedDate = date.slice(0, 10);

    return (
        <section className="details">
            {formattedDate === '' ? (
                <p>Click a marker above for permit details.</p>
            ) : (
                <div>
                    <h2>{props.streetNumber} {props.streetName} {props.streetSuffix} {props.unitNumber}</h2>
                    <div className="details-block">
                        <p><span className="underline">Status Date:</span> {formattedDate}</p>
                        <p><span className="underline">Permit Number:</span> {props.permitNumber}</p>
                        <p><span className="underline">Permit Status:</span> {props.permitStatus}</p>
                        <p><span className="underline">Description:</span> {props.permitDescription}</p>
                    </div>
                </div>
            )}
        </section>
    )
}