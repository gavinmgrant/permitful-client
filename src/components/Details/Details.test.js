import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Details from './Details';

describe(`Details component`, () => {

    it('renders a `.details`', () => {
        const wrapper = shallow(<Details statusDate='2011-08-25T00:00:00.000' />);
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});