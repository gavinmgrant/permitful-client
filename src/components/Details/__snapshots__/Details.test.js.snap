// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders a \`.details\` 1`] = `
<section
  className="details"
>
  <div
    className="details-result"
  >
    <h2>
       
       
    </h2>
    <div
      className="details-block"
    >
      <h3>
        <span
          className="bold"
        >
          Permit Number
        </span>
        : 
      </h3>
      <p>
        <span
          className="bold"
        >
          Status Date
        </span>
        : 
        2011-08-25
      </p>
      <p>
        <span
          className="bold"
        >
          Permit Status
        </span>
        : 
      </p>
      <p>
        <span
          className="bold"
        >
          Description
        </span>
        : 
      </p>
    </div>
  </div>
</section>
`;
